corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -343,20 +343,35 @@ hqDefine("cloudcare/js/formplayer/app", function () {
                 false
             );
         }
+
+        const pendingErrors = [];
+
         window.addEventListener(
-            'offline', function () {
-                showError(gettext("You are now offline. Web Apps is not optimized " +
-                    "for offline use. Please reconnect to the Internet before " +
-                    "continuing."), $("#cloudcare-notifications"));
-                $('.submit').prop('disabled', 'disabled');
-                $('.form-control').prop('disabled', 'disabled');
-            }
-        );
+            'offline',function () {
+                const offlineHandler = () => {
+                    showError(gettext("You are now offline. Web Apps is not optimized " +
+                        "for offline use. Please reconnect to the Internet before " +
+                        "continuing."), $("#cloudcare-notifications"));
+                    $('.submit').prop('disabled', 'disabled');
+                    $('.form-control').prop('disabled', 'disabled');
+                    pendingErrors.pop();
+                };
```

Comment:
I'm torn on the usefulness of an enclosed function here. Because this function isn't called anywhere else, its body could have just been written directly in the `if` block. It doesn't save any space, as the function is defined within this handler already. And for clarity, the handler kind of decides one level is appropriate for the `if` block, but another level is appropriate for the `else` block (i.e. `pendingErrors.pop()` is too low-level and should be hidden in the `offlineHandler`, but `pendingErrors.push()` is fine in the `else` block. This is pretty nitpicky, but I'm curious if this wouldn't just be better outside of this handler definition.